lib/rake/extensiontask.rb

```diff
@@ -158,26 +156,20 @@ def define_compile_tasks(for_platform = nil, ruby_ver = RUBY_VERSION)
       directory lib_path
       directory stage_binary_dir_path
 
-      directory File.dirname(siteconf_path)
-      # Set paths for "make install" destinations
-      file siteconf_path => File.dirname(siteconf_path) do
-        File.open(siteconf_path, "w") do |siteconf|
-          siteconf.puts "require 'rbconfig'"
-          siteconf.puts "require 'mkmf'"
-          siteconf.puts "dest_path = mkintpath(#{File.expand_path(lib_path).dump})"
-          %w[sitearchdir sitelibdir].each do |dir|
-            siteconf.puts "RbConfig::MAKEFILE_CONFIG['#{dir}'] = dest_path"
-            siteconf.puts "RbConfig::CONFIG['#{dir}'] = dest_path"
-          end
-        end
-      end
-
       # copy binary from temporary location to final lib
       # tmp/extension_name/extension_name.{so,bundle} => lib/
       task "copy:#{@name}:#{platf}:#{ruby_ver}" => [lib_path, tmp_binary_path, "#{tmp_path}/Makefile"] do
         # install in lib for native platform only
         unless for_platform
-          sh "#{make} install target_prefix=", chdir: tmp_path
+          expaned_lib_path = File.expand_path(lib_path)
```

Comment:
If we use absolute path, we need a special conversion on Windows that did by `mkintpath`:

https://github.com/ruby/ruby/blob/b5c74d548872388921402ff2db36be15e924a89b/lib/mkmf.rb#L1944-L1958

We don't need it when we use relative path.

Reply:
We also need to escape this path because this path may include a special character such as ` `.

Hmm. We may want to use `sh(*make_command_line, chdir: tmp_path)` with `Shellwords.shellsplit`:

```ruby
relative_lib_path = Pathname(lib_path).relative_path_from(tmp_path) 
make_command_line = Shellwords.shellsplit(make)
make_command_line << "install"
make_command_line << "sitearchdir=#{relative_lib_path}"
make_command_line << "sitelibdir=#{relative_lib_path}"
make_command_line << "target_prefix="
sh(*make_command_line, chdir: tmp_path)
```

Reply:
My initial thought of using relative path is that it would be easier to debug. However, the relative path is still debuggable and it will be better to avoid requiring `mkmkf`. (I just checked the `mkmkf`'s file size 😅 )

Also, thank you for pointing out the `Shellwords.shellsplit`, and it will be much more safer to use it!